src/request.c

```diff
@@ -83,22 +83,37 @@ void acceptGroupReport(uint32_t src, uint32_t group) {
         my_log(LOG_DEBUG, 0, "Should insert group %s (from: %s) to route table. Vif Ix : %d",
             inetFmt(group,s1), inetFmt(src,s2), sourceVif->index);
 
-        // If we don't have a whitelist we insertRoute and done
+        // If we don't have a black- and whitelist we insertRoute and done
         if(sourceVif->allowedgroups == NULL)
         {
             insertRoute(group, sourceVif->index, src);
             return;
         }
+
         // Check if this Request is legit on this interface
-        struct SubnetList *sn;
-        for(sn = sourceVif->allowedgroups; sn != NULL; sn = sn->next)
+        bool                 allow_list = 0;
+        bool                 block_list = 0;
+        struct SubnetList   *match = NULL;
+        struct SubnetList   *sn;
+
+        for(sn = sourceVif->allowedgroups; sn != NULL; sn = sn->next) {
+            // Check if there are black and/or whitelists
+            if (sn->allow)
+                allow_list = 1;
+            else
+                block_list = 1;
             if((group & sn->subnet_mask) == sn->subnet_addr)
-            {
-                // The membership report was OK... Insert it into the route table..
-                insertRoute(group, sourceVif->index, src);
-                return;
+                match = sn;
+        }
+
+        if((!allow_list && block_list &&
+          (match == NULL || (match != NULL && match->allow ))) ||
+          (allow_list && match != NULL && match->allow)) {
```

Comment:
This condition does not make sense... Condition `(match != NULL && match->allow )` is never true because at the beginning you check that `!allow_list` which is true only when `match->allow` is false.

Reply:
I'm clearly not seeing what you are seeing. Your guidiance is much appreciated.

After rewriting I come to the following
```
if((!allow_list && match == NULL) ||
  (allow_list && match != NULL && match->allow)
```

Does this makes sense?

Reply:
Yes, this logic looks to be usable. Just do not forget to document how whitelist/blacklist is evaluated.

Reply:
Evaluation has been documented in the manpage.